rules/DependencyInjection/Rector/Class_/ControllerGetByTypeToConstructorInjectionRector.php

```diff
@@ -0,0 +1,160 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\Symfony\DependencyInjection\Rector\Class_;
+
+use PhpParser\Node;
+use PhpParser\Node\Expr\ClassConstFetch;
+use PhpParser\Node\Expr\MethodCall;
+use PhpParser\Node\Stmt\Class_;
+use PHPStan\Reflection\ClassReflection;
+use Rector\Naming\Naming\PropertyNaming;
+use Rector\NodeManipulator\ClassDependencyManipulator;
+use Rector\PHPStan\ScopeFetcher;
+use Rector\PostRector\ValueObject\PropertyMetadata;
+use Rector\Rector\AbstractRector;
+use Rector\StaticTypeMapper\ValueObject\Type\FullyQualifiedObjectType;
+use Rector\Symfony\Enum\SymfonyClass;
+use Symplify\RuleDocGenerator\ValueObject\CodeSample\CodeSample;
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+
+/**
+ * @see \Rector\Symfony\Tests\DependencyInjection\Rector\Class_\ControllerGetByTypeToConstructorInjectionRector\ControllerGetByTypeToConstructorInjectionRectorTest
+ */
+final class ControllerGetByTypeToConstructorInjectionRector extends AbstractRector
+{
+    public function __construct(
+        private readonly ClassDependencyManipulator $classDependencyManipulator,
+        private readonly PropertyNaming $propertyNaming
+    ) {
+    }
+
+    public function getRuleDefinition(): RuleDefinition
+    {
+        return new RuleDefinition(
+            'From `$container->get(SomeType::class)` in controllers to constructor injection (step 1/x)',
+            [
+                new CodeSample(
+                    <<<'CODE_SAMPLE'
+use Symfony\Bundle\FrameworkBundle\Controller\Controller;
+
+final class SomeCommand extends Controller
+{
+    public function someMethod()
+    {
+        $someType = $this->get(SomeType::class);
+    }
+}
+CODE_SAMPLE
+                    ,
+                    <<<'CODE_SAMPLE'
+use Symfony\Bundle\FrameworkBundle\Controller\Controller;
+
+final class SomeCommand extends Controller
+{
+    public function __construct(private SomeType $someType)
```

Comment:
min php version 8.0 is needed for promotion property

Reply:
I see, it already handled at `$this->classDependencyManipulator->addConstructorDependency($node, $propertyMetadata)` that verify php version 👍